### PR TITLE
Improvements to pyQGIs code testing guidelines

### PR DESCRIPTION
**Pull request by SrNetoChan**
### Description

I have reordered some of your original content.

Goal:

Ticket: #

- [ ] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/documentation_guidelines/testing_guidelines.rst [hunk 1]
```diff
@@ -5,110 +5,143 @@
...
+  .. testoutput::
+
+      QGIS CRS ID: 3452
+      PostGIS SRID: 4326
+
+Grouping code snippets tests
+----------------------------
+
+For each rst document, the code snippets are tested sequentially, which means
+you can use one ``..testsetup::`` for all the following code snippets and that
```
Comment (DelazJ) [inline source/docs/documentation_guidelines/testing_guidelines.rst]:
```suggestion
you can use one ``.. testsetup::`` for all the following code snippets and that
```

File: source/docs/documentation_guidelines/testing_guidelines.rst [hunk 2]
```diff
@@ -5,110 +5,143 @@
...
+later snippets will use variables declared in previous ones.
```
Comment (DelazJ) [inline source/docs/documentation_guidelines/testing_guidelines.rst]:
```suggestion
latter snippets will use variables declared in previous ones.
```

File: source/docs/documentation_guidelines/testing_guidelines.rst [hunk 3]
```diff
@@ -5,110 +5,143 @@
...
+     assert crs.isValid()
 
-    print("QGIS CRS ID:", crs.srsid())
-    print("PostGIS SRID:", crs.srid())
+The doctest will pick each group snippets and run them independently.
 
-and then adding the expected results within the ``.. testoutput:: mysnippet``::
+If you don't declare any group, the code snippet will be added to the *default*
+group. If instead, you use ``*`` as a group name, the snippet will be used in
+all testing groups, something normally usefull to use in the test setup::
```
Comment (DelazJ) [inline source/docs/documentation_guidelines/testing_guidelines.rst]:
```suggestion
all testing groups, something normally useful to use in the test setup::
```
though I don't understand the part after the comma

File: source/docs/documentation_guidelines/testing_guidelines.rst [hunk 4]
```diff
@@ -5,110 +5,143 @@
...
+      print("QGIS CRS ID:", crs.srsid())
+      print("PostGIS SRID:", crs.srid())
+
+  .. testoutput::
+
+      QGIS CRS ID: 3452
+      PostGIS SRID: 4326
+
+Grouping code snippets tests
+----------------------------
```
Comment (DelazJ) [inline source/docs/documentation_guidelines/testing_guidelines.rst]: I always find weird to have a single sub section under a section

File: source/docs/documentation_guidelines/testing_guidelines.rst [hunk 5]
```diff
@@ -5,110 +5,143 @@
...
-you have to follow some rules to enable automatic testing of code snippets.
+If you are planning to add or update some chapters of the
+:ref:`PyQGIS-Developer-Cookbook`, then you should follow some rules to enable
+automatic testing of the code snippets.
 
-Testing is really important because we can check automatically the quality of the
-code: if the code is not good, tests will fail meaning that the snippet has
-some errors.
+Testing is really important because allows checking automatically the quality of
+the code. During testing, if the code is not good or outdated, tests will fail,
```
Comment (DelazJ) [inline source/docs/documentation_guidelines/testing_guidelines.rst]:
```suggestion
the code. During testing, if the code is not good or is outdated, tests will fail,
````
I understood "is not outdated"

File: source/docs/documentation_guidelines/testing_guidelines.rst (hunk 4 above)
Comment (SrNetoChan) [reply source/docs/documentation_guidelines/testing_guidelines.rst]: Any suggestions? we can simply remove it?

**Comment by DelazJ** [comment]
Can this PR be closed? Not needed anymore and too much noise. Thanks

**Pull request closed without merging**